ndbench-core/src/main/java/com/netflix/ndbench/core/DataBackfill.java

```diff
@@ -79,22 +90,20 @@ public void backfillAsync(final NdBenchAbstractClient<?> client) {
     private void backfillAsync(final NdBenchAbstractClient<?> client, final BackfillOperation backfillOperation) {
         stop.set(false);
 
-        final int numThreads = config.getNumBackfill();
-        final int backFillStartKey = config.getBackfillStartKey();
-        final int numKeysPerThread = (config.getNumKeys() - backFillStartKey) / numThreads;
-
-        logger.info("NUM THREADS: " + numThreads);
-        logger.info("NUM KEYS: " + config.getNumKeys());
-        logger.info("NUM KEYS PER TH: " + numKeysPerThread);
+        //Default #Cores*4 so that we can keep the CPUs busy even while waiting on I/O
+        final int numThreads = Runtime.getRuntime().availableProcessors() * 4;
 
         initThreadPool(numThreads);
 
+        List<Pair<Integer, Integer>> keyRanges = getKeyRangesPerThread(numThreads,
```

Comment:
initThreadPool(numThreads) on line 96 creates a thread pool with numThreads+1 threads. yet getKeyRangesPerThread only divides by numThreads and not numThreads+1. comment on why?

Reply:
That 1 extra thread is for the backfill status poller.